kit/src/main/java/com/oracle/javafx/scenebuilder/kit/editor/DocumentationUrls.java

```diff
@@ -0,0 +1,247 @@
+ * essential JavaFX developer documentation.
+ */
+public class DocumentationUrls {
+
+    private static final String UNKNOWN = "unknown";
+
+    private static DocumentationUrls instance = null;
+
+    public static DocumentationUrls getInstance() {
+        if (instance == null) {
+            instance = new DocumentationUrls();
+        }
+        return instance;
+    }
+
+    /**
+     * Verifies if the desired {@link DocumentationItem} is available.
+     * 
+     * @param item {@link DocumentationItem} (usually an URL)
+     * @return true when available in corresponding resource
+     */
+    public static boolean isAvailable(DocumentationItem item) {
+        return getInstance().getOptionalUrl(item).isPresent();
+    }
+
+    /**
+     * Detects the used JavaFX version from system properties using
+     * {@code System.getProperty("javafx.version")}. In case this property is
+     * undefined, the String "unknown" is provided.
+     * 
+     * @return JavaFX version number or "unknown" when property is undefined
+     */
+    protected static String getJavaFxVersion() {
+        return getJavaFxVersion(System.getProperties());
+    }
+
+    /**
+     * Detects the used JavaFX version from given properties instance using property
+     * key {@code "javafx.version"}. In case this property is undefined, the String
+     * "unknown" is provided.
+     * 
+     * @return JavaFX version number or "unknown" when property is undefined
+     */
+    protected static String getJavaFxVersion(Properties systemProperties) {
+        Objects.requireNonNull(systemProperties);
+        String fxVersion = systemProperties.getProperty("javafx.version");
+        if (fxVersion == null) {
+            return UNKNOWN;
+        }
+        return fxVersion;
+    }
+
+    /**
+     * @param version JavaFX version string
+     * @return Provides the major version of the given version String.
+     */
+    protected static String getMajorJavaFxVersion(String version) {
+        if (version == null || version.isBlank()) {
+            return UNKNOWN;
+        }
+        int firstDot = version.indexOf('.');
+        if (firstDot > 1) {
+            return version.substring(0, firstDot);
+        } else {
+            return version;
+        }
+    }
+
+    private final Map<DocumentationItem, String> configuredUrls;
+
+    private final String javaFxVersion;
+
+    private DocumentationUrls() {
+        this(getJavaFxVersion(), "documentation_urls.properties");
+    }
+
+    DocumentationUrls(String javaFxVersion, String resourceName) {
+        Objects.requireNonNull(resourceName, "resourceName must not be null");
+        this.configuredUrls = new EnumMap<>(DocumentationItem.class);
+        this.javaFxVersion = Objects.requireNonNull(javaFxVersion, "javaFxVersion must not be null");
+        loadProperties(resourceName);
+    }
+
+    private void loadProperties(String resourceName) {
```

Comment:
This seems like too much work to externalise these links. If we hardcode them like DocumentationItem enum, a lot of the code and files would just go I think. Something like:

docitem.defaultURL which we already have and
docitem.url to replace, this is where the properties file data goes.

Reply:
The approach with the properties file has been reworked. `DocumentationUrls` is now the enum type which provides the URLs by name. It also does the placeholder resolution to have the used JavaFX version when needed.